aspnetcore/web-api/index.md

```diff
@@ -167,7 +167,64 @@ To make automatic and custom responses consistent, call the <xref:Microsoft.AspN
 
 ### Log automatic 400 responses
 
-See [How to log automatic 400 responses on model validation errors (dotnet/AspNetCore.Docs#12157)](https://github.com/dotnet/AspNetCore.Docs/issues/12157).
+You can add logging by utilizing [InvalidModelStateResponseFactory](https://docs.microsoft.com/dotnet/api/microsoft.aspnetcore.mvc.apibehavioroptions.invalidmodelstateresponsefactory). By default, [InvalidModelStateResponseFactory](https://docs.microsoft.com/dotnet/api/microsoft.aspnetcore.mvc.apibehavioroptions.invalidmodelstateresponsefactory) uses [ProblemDetailsFactory](https://docs.microsoft.com/dotnet/api/microsoft.aspnetcore.mvc.infrastructure.problemdetailsfactory) to construct an instance of [ValidationProblemDetails](https://docs.microsoft.com/dotnet/api/microsoft.aspnetcore.mvc.validationproblemdetails). Here is an example of code to be used in `Startup.ConfigureServices` that adds logging and preserves the original behavior:
+
+```csharp
+services.AddControllers().ConfigureApiBehaviorOptions(options =>
+{
+    // If preserving of the original behavior is desired, get a reference to the delegate.
+    var builtInFactory = options.InvalidModelStateResponseFactory;
+
+    options.InvalidModelStateResponseFactory = context =>
+    {
+        // As an example, we will create a logger with the full name of the action method as the category.
+        var logger = GetLoggerInstance(context.HttpContext.RequestServices, context.ActionDescriptor.DisplayName);
+
+        // Log accordingly.
+
+        // By using the original delegate, we preserve the default behavior.
+        // Alternatively, you can take full control and simply construct the ValidationProblemDetails object yourself,
+        // or even use a custom object.
+        var result = builtInFactory(context);
+
+        // Example of manually constructing a ValidationProblemDetails object:
```

Comment:
Is this example of manually constructing a ValidationProblemDetails too much?

Reply:
https://github.com/MohammadAsadi1372/MohammadAsadi.MasterService

Reply:
I feel like we should avoid linking to the issue, hence why we're giving every bit of information the end-user might require for automatic logging of 400 requests in the document itself, however I'm fine with completely dropping the example for instancing a `ValidationProblemDetails` object. Alternatively, we can find it's place outside of the code example, instead of linking to the issue?

Reply:
Ok, let's just drop it for now then. I'm not sure where else it would live right now, so perhaps it's something we could come back to in the future.